packages/backend/src/util/__tests__/handshakeUtils.test.ts

```
@@ -0,0 +1,152 @@
+    });
+
+    it('should enhance URLs with redirect_url containing oauth callback', () => {
+      const callbackUrl = 'https://app.example.com/oauth_callback';
+      const originalUrl = `https://provider.com/oauth/authorize?redirect_url=${encodeURIComponent(callbackUrl)}`;
+      const enhancedUrl = enhanceOAuthRedirectUrl(originalUrl, mockAuthenticateContextWithNonce);
+
+      const url = new URL(enhancedUrl);
+      const enhancedCallbackUrl = url.searchParams.get('redirect_url');
+      expect(enhancedCallbackUrl).toBeDefined();
+
+      const callbackUrlObj = new URL(enhancedCallbackUrl!);
```

ESLint

[error] Forbidden non-null assertion.

(@typescript-eslint/no-non-null-assertion)